meta-dts-distro/recipes-dts/dts/dts/dts-functions.sh

```diff
@@ -741,3 +690,211 @@ verify_artifacts() {
+      BINARY_HAS_ME=1
+    fi
+    # Check for aa55 signature at ME offset + 4096 (new ME)
+    if [ $(tail -c +$((0x$ME_OFFSET + 4097)) $1|head -c 2|xxd -ps) == "aa55" ]; then
+      BINARY_HAS_ME=1
+    fi
+  fi
+}
+
+check_if_me_disabled() {
+
+  ME_DISABLED=0
+
+  if [ $BOARD_HAS_ME_REGION -eq 0 ]; then
+    # No ME region
+    ME_DISABLED=1
+    return
+  fi
+
+  # Check if HECI present
+  # FIXME: what if HECI is not device 16.0?
+  if [ -d /sys/class/pci_bus/0000:00/device/0000:00:16.0 ]; then
+    # Check ME Current Operation Mode at offset 0x40 bits 19:16
+    ME_OPMODE="$(setpci -s 00:16.0 42.B 2> /dev/null | cut -c2-)"
+    if [ $ME_OPMODE == "0" ]; then
+      echo "ME is not disabled"  >> $ERR_LOG_FILE
+      return
+    elif [ $ME_OPMODE == "2" ]; then
+      echo "ME is disabled (HAP/Debug Mode)"  >> $ERR_LOG_FILE
+      ME_DISABLED=1
+      return
+    elif [ $ME_OPMODE == "3" ]; then
+      echo "ME is soft disabled (HECI)"  >> $ERR_LOG_FILE
+      ME_DISABLED=1
+      return
+    elif [ $ME_OPMODE == "4" ]; then
+      echo "ME disabled by Security Override Jumper/FDOPS"  >> $ERR_LOG_FILE
+      ME_DISABLED=1
+      return
+    elif [ $ME_OPMODE == "5" ]; then
+      echo "ME disabled by Security Override MEI Message/HMRFPO"  >> $ERR_LOG_FILE
+      ME_DISABLED=1
+      return
+    elif [ $ME_OPMODE == "5" ]; then
+      echo "ME disabled by Security Override MEI Message/HMRFPO"  >> $ERR_LOG_FILE
+      ME_DISABLED=1
+      return
+    elif [ $ME_OPMODE == "7" ]; then
+      echo "ME disabled (Enhanced Debug Mode) or runs Ignition FW"  >> $ERR_LOG_FILE
+      ME_DISABLED=1
+      return
+    else
+      print_warning "Unknown ME operation mode, assuming enabled."
+      echo "Unknown ME operation mode, assuming enabled."  >> $ERR_LOG_FILE
+      return
+    fi
+  else
+    # If we are running coreboot, check for status in logs
+    cbmem -1 | grep -q "ME is disabled" && ME_DISABLED=1 && return # HECI (soft) disabled
+    cbmem -1 | grep -q "ME is HAP disabled" && ME_DISABLED=1 && return # HAP disabled
+    # TODO: If proprietary BIOS, then also try to check SMBIOS for ME FWSTS
+    # BTW we could do the same in coreboot, expose FWSTS in SMBIOS before it
+    # gets disabled
+    print_warning "Can not determine if ME is disabled, assuming enabled."
+    echo "Can not determine if ME is disabled, assuming enabled."  >> $ERR_LOG_FILE
+  fi
+}
+
+force_me_update() {
+    echo
+    print_warning "Flashing ME when not in disabled state may cause unexpected power management issues."
+    print_warning "Recovering from such state may require removal of AC power supply and resetting CMOS battery."
+    print_warning "Keeping an older version of ME may cause a CPU to perform less efficient, e.g. if upgraded the CPU to a newer generation."
+    print_warning "You have been warned."
+  while : ; do
+    echo
+    read -r -p "Force the flashing without updating ME? (Y|n) " OPTION
```

Comment:
Not sure if I understand this function fully. Should this say "Force the flashing without *disabling* ME"?

Reply:
No, it asks to flash the BIOS update without flashing/updating ME. YOu may still want to flash just the BIOS region right?

`Skip ME update and proceed with BIOS/fiurmware flashing?`

sounds better?

Reply:
https://github.com/Dasharo/meta-dts/pull/65/commits/220f636c1d0d9c0f44f9a2ccb82121dcb94df17d#diff-83430a524e32e99ff8921d9df707448a35c7657511ebfe205f631911ab9c4ac1R827